core/win32/syscall.c

```diff
@@ -1853,8 +1853,15 @@ presys_SetContextThread(dcontext_t *dcontext, reg_t *param_base)
     /* FIXME : we are going to read and write to cxt, which may be unsafe */
     ASSERT(tid != 0xFFFFFFFF);
     LOG(THREAD, LOG_SYSCALLS|LOG_THREADS, IF_DGCDIAG_ELSE(1, 2),
-        "syscall: NtSetContextThread handle="PFX" tid=%d cxt->Xip="PFX"\n",
-        thread_handle, tid, cxt->CXT_XIP);
+        "syscall: NtSetContextThread handle="PFX" tid=%d cxt->Xip="PFX" flags="PFX"\n",
+        thread_handle, tid, cxt->CXT_XIP, cxt->ContextFlags);
+    if (get_thread_id() == tid) {
+        /* Simple case when called on own thread. */
+        /* FIXME : we should handle theses flags. */
+        ASSERT(!TEST(CONTEXT_CONTROL, cxt->ContextFlags) &&
```

Comment:
Should be ASSERT_NOT_IMPLEMENTED: regular ASSERT implies this should never happen and that if it does it's some internal logic error, whereas this is a legitimate app input.